test/fixtures/correct.js

```diff
@@ -230,6 +230,13 @@ const quotes2 = `foo`;
 noop(quotes1);
 noop(quotes2);
 
+// `require-atomic-updates`.
```

Comment:
Missing the incorrect version of this rule.

Reply:
There is no incorrect version because this is disabling an unwanted rule that was added in `eslint@6.0.0`. This test shows the incorrect behaviour, and asserts it's not throwing a linter error.

Reply:
This does not seem like the incorrect behavior :thinking:

From the docs, I get the sense that this rule is trying to prevent calling await directly after reading from a variable, to avoid the read variable to change while awaiting, and then the update is done with the stale value for the variable.

Reply:
This is also incorrect behaviour, and will fail if we reenable the rule. That's because we're mutating `foo` inside of an async function, which could potentially cause trouble for multiple function calls, where order is not guaranteed.

The main motivator to disable this rule was because in new projects that used `eslint@6` we were getting this error when doing:

```js
async grpcMethod(ctx) {
  ctx.res = mask('foo');
}
```

Reply:
A lot of people have been complaining that this rule is in fact overly aggressive and should be disabled until it is better defined.

Seeing that this rule right now is not working correctly, we should indeed disable it, but I would suggest adding a comment in the line that disables this rule that mentions why it needs to be disabled and that it can/should be enabled when it is fixed.

Reply:
It doesn't seem like we do that sort of comments in that file. So instead I've created an issue to help us track this: https://github.com/uphold/eslint-config-uphold/issues/17.

Reply:
Sure, visibility is the end goal.